src/values/ObjectValue.js

```diff
@@ -585,8 +586,15 @@ export default class ObjectValue extends ConcreteValue {
       !this.isSimpleObject() ||
       (P.mightNotBeString() && P.mightNotBeNumber() && !P.isSimpleObject())
     ) {
-      AbstractValue.reportIntrospectionError(P, "TODO: #1021");
-      throw new FatalError();
+      // if P is an abstract value that we don't know about, but wee're in pure scope
+      // then if the object is **not** partial, but is simple, then we can
```

Comment:
I don't think we need to be concerned that an object is partial or not. Only that it is simple, because we are emitting runtime code and we need to be sure that fetching the property will not cause havoc.